dockerfiles/pr-check/Dockerfile

```diff
@@ -6,11 +6,31 @@
 
 FROM library/centos:centos7
 
-USER root
+# Install dependencies; note rh-nodejs8 is required for dashboard build to succeed
+RUN yum update --assumeyes \
+ && yum install --assumeyes \
+        docker \
+        tree \
+        git \
+        patch \
+        pcp \
+        bzip2 \
+        golang \
+        make \
+        java-1.8.0-openjdk \
+        java-1.8.0-openjdk-devel \
+        centos-release-scl \
+ && yum install --assumeyes \
+        rh-maven33 \
+        rh-nodejs8 \
+ && yum clean all \
+ && rm -rf /var/cache/yum
+
+# Bower update step won't work for root user
+RUN useradd -ms /bin/bash newuser
```

Comment:
Nice :) We should still try to keep the RUN commands minimized into only single call because every time RUN is called, it creates a new docker layer, thus increasing the docker image size

Reply:
Creating new docker layers isn't a big problem if the `RUN` commands affect different files; the issue arises when e.g. files are created in one `RUN` command and then deleted in another. `docker inspecting` images with and without the `useradd` step being combined with the previous `RUN` gives
- `"Size": 2304547111` (2304.55MB) for current PR
- `"Size": 2304254909` (2304.25MB) for dockerfile with useradd combined with previous command.

I also think this is more maintainable and makes more sense, but I'm happy to make the change if you prefer.

Reply:
:+1: in that case. Thank you for explaining the inner workings of docker :) I think we are ready for merge then.